### PR TITLE
[WIP] Work on documentation

### PR DESCRIPTION
**Pull request by FardaleM**
I may add more doc in the future. Maybe, the best plan is to merge this just before the next release.

### CODE REVIEW

File: src/core/CCResult.mli [hunk 1]
```diff
@@ -188,6 +188,9 @@ end
 (** {2 Collections} *)
 
 val map_l : ('a -> ('b, 'err) t) -> 'a list -> ('b list, 'err) t
+(** [map_l f l] apply [f] on every element of [l], and, in case of success for every
+    element, return [Ok (List.map f l)]. Otherwise, it fails, and the first error is
+    return.*)
```
Comment (c-cube) [inline src/core/CCResult.mli]: "returned" :slightly_smiling_face: 
also, it's not really `Ok (List.map f l)` but the list of all the `Ok`-values, right?

Comment (FardaleM) [reply src/core/CCResult.mli]: It is equivalent to `Ok (List.map f l)` but it does not call List.map. It is the list of unwrap Ok-values. But I don't know how to write it accurately and concisely.

Comment (pmetzger) [reply src/core/CCResult.mli]: Explain it verbosely (I don't currently understand what the function does) and I will edit it down.

Comment (FardaleM) [reply src/core/CCResult.mli]: I pushed a new version of this.

File: src/core/CCResult.mli [hunk 2]
```diff
@@ -188,6 +188,9 @@ end
 (** {2 Collections} *)
 
 val map_l : ('a -> ('b, 'err) t) -> 'a list -> ('b list, 'err) t
+(** [map_l f \[a1; ...; an\]] applies function f to a1, ..., an , and, in case of
+    success for every elements, return the list of [Ok]-value.
```
Comment (pmetzger) [inline src/core/CCResult.mli]: Small note: `element` should be singular (to agree grammatically with `every`.)

Comment (pmetzger) [reply src/core/CCResult.mli]: Oh, and `returns` should be plural.

**Pull request merged**